### PR TITLE
Migrate the DB config from yaml template to viper

### PR DESCRIPTION
**Pull request by chrisgilmerproj**
## Description

Pop currently uses a YAML config file that is a go template to load environment variables. This indirect approach should be changed to manually creating a connection on startup while using viper to capture the configuration values.

## Reviewer Notes

I'm a bit worried that I'm overly verbose with error messages and would love input about it.

## Setup

Start the server and the app.  You should be able to get any db query to work.

## Code Review Verification Steps

* [x] Request review from a member of a different team.
* [x] Have the Pivotal acceptance criteria been met for this change?

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/161272729) for this change

### CODE REVIEW

**Comment by pjdufour-truss** [comment]
Could you add more context to your zap errors?  I think a connection can be output as a url string?  And then maybe manually try a bad set of connection details and see what happens.

**Comment by pjdufour-truss** [comment]
Or maybe just the username, host, port, and db separately, so we don't add password to logs by accident?

**Comment by chrisgilmerproj** [comment]
Yeah, it should be hard to add more context there.  Also, I noticed that all the other binaries in `./bin/` use the `database.yml` file.  Should I remove those references and also use pflags/viper and my new `initDatabase` fn?  I'm mostly worried that they will be forgotten and I'd love to keep everything consistent.  The change is just a lot of grunt work and it might blow up the size of this PR.

**Comment by chrisgilmerproj** [comment]
This appears to have gone up to `experimental` just fine and I can make changes to the DB there.  

I have one more change before I'd be ready to merge this.

**Comment by pjdufour-truss** [comment]
Could you make the port an Int config variable and then use `strconv.Itoa` to pass it to pop?  We want to leverage the built-in validation with Viper as much as possible (before we start adding our own on top of it).

**Comment by chrisgilmerproj** [comment]
I think I'll be merging this Monday just to make sure we have everyone around if there's an issue.

File: cmd/webserver/main.go [hunk 1]
```diff
@@ -231,6 +240,66 @@ func initRealTimeBrokerService(v *viper.Viper, logger *zap.Logger) (*iws.RealTim
...
+	err := dbConnectionDetails.Finalize()
+	if err != nil {
+		logger.Error("Failed to finalize DB connection details", zap.Error(err))
+		return nil, err
+	}
+
+	// Set up the connection
+	connection, err := pop.NewConnection(&dbConnectionDetails)
+	if err != nil {
+		logger.Error("Failed create DB conection", zap.Error(err))
```
Comment (chrisgilmerproj) [inline cmd/webserver/main.go]: Fix typo! Thanks @kilbergr !

File: cmd/webserver/main.go [hunk 2]
```diff
@@ -231,6 +240,66 @@ func initRealTimeBrokerService(v *viper.Viper, logger *zap.Logger) (*iws.RealTim
...
+		// environment as development, and it's extra confusing to have to swap env
+		// variables before running tests.
+		dbName = "test_db"
+	} else if env == "container" {
+		// Require sslmode for containers
+		dbOptions["sslmode"] = "require"
+	}
+
+	// Construct a safe URL and log it
+	s := "postgres://%s:%s@%s:%s/%s?sslmode=%s"
```
Comment (chrisgilmerproj) [inline cmd/webserver/main.go]: port is an integer!

Comment (chrisgilmerproj) [reply cmd/webserver/main.go]: Oh, turns out its safe because I modified it above to be a string already. Whew!

**Pull request merged**